aspnet/fundamentals/owin.rst

```diff
@@ -48,53 +48,57 @@ You can of course configure other actions to take place within the OWIN pipeline
-    });
+      {
+          // do something before
+          return OwinHello;
+          // do something after
+      });
   });
 
 .. note:: The OWIN support in ASP.NET Core is an evolution of the work that was done for the `Katana project <http://katanaproject.codeplex.com/>`_. Katana's ``IAppBuilder`` component has been replaced by ``IApplicationBuilder``, but if you have existing Katana-based middleware, you can use it within your ASP.NET Core application through the use of a bridge, as shown in the `Owin.IAppBuilderBridge example on GitHub <https://github.com/aspnet/Entropy/tree/master/samples/Owin.IAppBuilderBridge>`_. 
 
 Using ASP.NET Hosting on an OWIN-based server
 ---------------------------------------------
 
-OWIN-based servers can host ASP.NET applications, since ASP.NET conforms to the OWIN specification. One such server is `Nowin <https://github.com/Bobris/Nowin>`_, a .NET OWIN web server. In the sample for this article, I've included a very simple project that references Nowin and uses it to create a simple server capable of self-hosting ASP.NET Core.
+OWIN-based servers can host ASP.NET applications. One such server is `Nowin <https://github.com/Bobris/Nowin>`_, a .NET OWIN web server. In the sample for this article, I've included a project that references Nowin and uses it to create an ``IServer`` capable of self-hosting ASP.NET Core.
 
-.. literalinclude:: owin/sample/src/NowinSample/NowinServerFactory.cs
-  :emphasize-lines: 13,19,22,27,41
+.. literalinclude:: owin/sample/src/NowinSample/NowinServer.cs
+  :emphasize-lines: 15
   :linenos:
   :language: c#
 
-``IServerFactory`` is an interface that requires an ``Initialize`` and a ``Start`` method. Initialize must return an instance of :dn:iface:`~Microsoft.AspNetCore.Http.Features.IFeatureCollection`, which we populate with a ``INowinServerInformation`` that includes the server's name (the specific implementation may provide additional functionality). In this example, the ``NowinServerInformation`` class is defined as a private class within the factory, and is returned by ``Initialize`` as required.
+``IServer`` is an interface that requires an ``Features`` property and a ``Start`` method.
 
-``Initialize`` is responsible for configuring the server, which in this case is done through a series of fluent API calls that hard code the server to listen for requests (to any IP address) on port 5000. Note that the final line of the fluent configuration of the ``builder`` variable specifies that requests will be handled by the private method ``HandleRequest``.
+``Start`` responsible for configuring and starting the server, which in this case is done through a series of fluent API calls that set addresses parsed from the IServerAddressesFeature. Note that the fluent configuration of the ``_builder`` variable specifies that requests will be handled by the ``appFunc`` defined earlier in the method. This ``Func`` is called on each request to process incoming requests.
```

Comment:
>This ``Func`` is called on each request to process incoming requests. 

Is there anyway to simplify that?

Can we just say 
This ``Func`` is called on each request.